src/neighbors.cc

```diff
@@ -1453,7 +1453,7 @@ dump_peer_options(StoreEntry * sentry, CachePeer * p)
         storeAppendPrintf(sentry, " ttl=%d", p->mcast.ttl);
 
     if (p->connect_timeout_raw > 0)
-        storeAppendPrintf(sentry, " connect-timeout=%d", (int)p->connect_timeout_raw);
+        storeAppendPrintf(sentry, " connect-timeout=%ld", p->connect_timeout_raw);
```

Comment:
I am not excited about this change because it makes the problem more difficult to see rather than fixing it: `long int` is [guaranteed](https://en.cppreference.com/w/cpp/language/types) to have 32 bits, not 64 bits as this fix (description) may want us to believe. I made the same mistake in commit 5ef3cbb, but I did not have Coverity results to help me see the light.

Want a better minimal fix? Cast to and printf() int64_t.

Want to stop kicking the can down the road? Convert the entire dump_peer_options() to use PackableStream instead. See SBufStatsAction::dump(), dump_refreshpattern(), Mem::Stats(), and similar functions for examples.

Reply:
done